runtime/src/serde_snapshot/newer.rs

```diff
@@ -39,8 +39,7 @@ struct DeserializableVersionedBank {
     ns_per_slot: u128,
     genesis_creation_time: UnixTimestamp,
     slots_per_year: f64,
-    #[allow(dead_code)]
-    unused: u64,
```

Comment:
well that's convenient!  No issue with newer code reading an older snapshot and decoding `accounts_data_len` as 0?

Reply:
> No issue with newer code reading an older snapshot and decoding `accounts_data_len` as 0?

Correct, no issue. The main reason is that `Bank::new_from_fields()` doesn't use the snapshot (i.e. `fields`) to set its `accounts_data_len` field yet [1]; that value is calculated during `generate_index()` [2] and passed up as a parameter to `new_from fields()` [3].

So this PR will start _writing_ real-ish values into snapshots, but not _reading_ that value yet. PR #23744 will start reading the value (and I elaborate a bit more on the safety of `fields.accounts_data_len` too).

[1] https://github.com/solana-labs/solana/blob/6b0d34d70dc08beb3d94cc8e30423883de73646b/runtime/src/bank.rs#L2007

[2] https://github.com/solana-labs/solana/blob/6b0d34d70dc08beb3d94cc8e30423883de73646b/runtime/src/serde_snapshot.rs#L349-L357

[3] https://github.com/solana-labs/solana/blob/6b0d34d70dc08beb3d94cc8e30423883de73646b/runtime/src/bank.rs#L1930-L1939